src/coreclr/jit/ifconversion.cpp

```diff
@@ -668,7 +668,7 @@ bool OptIfConversionDsc::optIfConvert()
         }
 
         // We may be inside an unnatural loop, so do the expensive check.
-        if (m_comp->optReachable(m_finalBlock, m_startBlock, nullptr))
+        if (m_comp->compHasBackwardJump && m_comp->optReachable(m_finalBlock, m_startBlock, nullptr, /*budget*/ 8))
```

Comment:
For now I would rather we avoid changing the heuristic here and instead use an if-conversion wide (but much larger) budget. E.g. `int reachabilityBudget = 100000;` initialized at the beginning of `optIfConversion` and then used here repeatedly.